### DISCUSSION TITLE
Question about texture applied in Big Bank

### DISCUSSION THREAD

**Discussion by miniwolf** [diesel-modding/BeardLib-Editor#763] [Q&A] [answered]
I have a question about the roof on Big Bank (Dentist). In the material config I see mat_roof_gravel:
```
       <material name="mat_roof_gravel" render_template="generic:BLEND_DIFFUSE:BLEND_MASK_SEPERATE:BLEND_NORMAL:DIFFUSE_TEXTURE:NORMALMAP:VERTEX_COLOR" version="2">
		<diffuse_texture file="units/pd2_dlc2/architecture/gov_c_ext/gov_c_ext_roof_gravel_b_df"/>
		<bump_normal_texture file="units/pd2_dlc2/architecture/gov_c_ext/gov_c_ext_roof_gravel_b_nm"/>
		<diffuse_layer0_texture file="units/pd2_dlc2/architecture/gov_c_ext/gov_c_ext_roof_gravel_a_df"/>
		<normal_layer0_texture file="units/pd2_dlc2/architecture/gov_c_ext/gov_c_ext_roof_gravel_a_nm"/>
		<diffuse_layer1_texture file="units/pd2_dlc2/architecture/gov_c_ext/gov_c_ext_roof_gravel_mask_df"/>
	</material>
```

I tried to replicate this by extracting the roof object and importing it in Blender. However, when I mix the two diffuse and normal textures together and apply the mask, I do not get the effect seen in the game. There seems to be some additional mask added somehow, to get the mask to be none repeating in those big areas.

Is there any information I am missing?

**Comment by Xeletron** [comment, marked as answer]
I reccomend using the Diesel material groups to help you make things look somewhat closer to the game https://modworkshop.net/mod/26591
in this case you're going to need the PD2 DF:NM:VTxBlendMasked surface group
![image](https://user-images.githubusercontent.com/13988064/221532445-d4d13102-7529-425f-bd0f-f1eccf748252.png)
After you have all that set up you need to connect the Color Attribute alpha to the surface groups 'blend mask' and 'vertex color alpha' nodes.
![image](https://user-images.githubusercontent.com/13988064/221557570-f7982356-3756-409c-b6aa-24babc2acf6f.png)
It still wont be an exact match to how it looks in game but it will look more comperable.

Also reccomend changing your image texture's alpha to premultiplied or channel packed to make them not look really dark.
![image](https://user-images.githubusercontent.com/13988064/221534141-961ad226-e3dc-402e-bc17-1f4162471d26.png)

**Comment by miniwolf** [comment]
Ohh that is super interesting @Xeletron. That gives me something to play around with. I will report back if I get the result I am looking for.

**Comment by miniwolf** [comment]
So following up, this showed me the mix happens due to the color attribute. It is super interesting!
